src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSchemaJsonWriter.cs

```diff
@@ -463,7 +463,7 @@ internal override void WriteDecimalTypeAttributes(IEdmDecimalTypeReference refer
             // The value of $Scale is a number or a string with one of the symbolic values floating or variable.
             // TODO: the symbolic values floating or variable is not supported now.
             // Absence of $Scale means variable.
-            this.jsonWriter.WriteOptionalProperty("$Scale", reference.Scale, CsdlConstants.Default_Scale);
+            this.jsonWriter.WriteOptionalProperty("$Scale", reference.Scale);

```

Comment:
Why this change?
Have we stopped writing `scale="variable"`?

Reply:
Why don't you introduce the condition you introduced in the xml writer?

Reply:
Hmm those are good questions.

The reason for this change was that `this.WriteOptionalProperty` overload that was used before accepted a default value as the last argument. It is implemented as follows: 
```c#
public static void WriteOptionalProperty(this Utf8JsonWriter writer, string name, int? value, int defaultValue)
{
    EdmUtil.CheckArgumentNull(writer, nameof(writer));

    if (value != null && value.Value != defaultValue)
    {
        writer.WritePropertyName(name);
        writer.WriteNumberValue(value.Value);
    }
}
```
Now, the `CsdlConstants.Default_Scale` is no longer an `int` but a nullable `int?`, and so can't be passed to this overload. But since the default scale is `null`, it's not necessary to use this overload, we can use the following one:
```c#
public static void WriteOptionalProperty(this Utf8JsonWriter writer, string name, int? value)
{
    EdmUtil.CheckArgumentNull(writer, nameof(writer));

    if (value != null)
    {
        writer.WritePropertyName(name);
        writer.WriteNumberValue(value.Value);
    }
}
```

Since the default scale is `null`, the behaviour of both methods is the same because they both write nothing if the input `value` is `null` regardless of what the default value is. So this change does not affect the existing behaviour.

Reply:
To answer the other question, why I didn't introduce the condition I introduced in the XML writer (i.e. to check whether `ShouldWriteDefaultScale` is set to true) is because prior to this PR, the default behaviour when writing scale was different between XML and JSON CSDL. In XML 0 was the default scale, and having a null scale would end up writing it to the XML. In JSON, null was the default scale and it was not written out to the CSDL document. So setting the default scale to null would not have any breaking changes for JSON CSDLs, it only affects XML CSDL.

I'm pushing new commits which change the XML logic to always write the scale value regardless of default values, this would have even fewer chances of breaking changes.